src/gamestate/system_state.cpp

```diff
@@ -541,403 +537,12 @@ void state::render() { // called to render the frame may (and should) delay retu
-					provinces_to_reduce_weight.push_back(current_location);
-				} else {
-					provinces_to_maintain.push_back(current_location);
-				}
-				l += 1;
-
-				for(auto adj : world.province_get_province_adjacency(current_location)) {
-					auto other = adj.get_connected_provinces(adj.get_connected_provinces(0) == current_location ? 1 : 0);
-
-					if(std::find(province_queue.begin(), province_queue.end(), other) == province_queue.end()) {
-						province_queue.push_back(other);
-						r += 1;
-					}
-				}
-			}
-
-			// if l < r then there is a vacant province and we had stopped early
-			// so try to fill the vacant location
-			if(l < r) {
-				auto target_location = province_queue[l];
-
-				/*
-				for(uint32_t i = 0; i < military_definitions.unit_base_definitions.size(); ++i) {
-					regiments_distribution[i] = 0.f;
-				}
-
-				//recalculate distribution
-				float total = 0.f;
-				for(dcon::regiment_id regiment : army_group.land_regiments) {
-					auto regiment_type = world.regiment_get_type(regiment);
-					auto status = army_group.regiment_status[regiment.index()];
-					if(status == army_group_regiment_status::idle) {
-						regiments_distribution[regiment_type.index()] += 1.f;
-						total += 1.f;
-					}
-				}
-
-				bool success = fill_province_up_to_supply_limit(&army_group, target_location, regiments_distribution, army_group_regiment_status::idle, army_group_regiment_status::awaiting_orders);
-				if(success) {
-					break;
-				}
-				*/
-
-				for(auto current_regiment : army_group.land_regiments) {
-					auto army = world.regiment_get_army_from_army_membership(current_regiment);
-					auto current_location = world.army_get_location_from_army_location(army);
-					auto current_path = world.army_get_path(army);
-					//auto status = army_group.regiment_status[current_regiment.index()];
-					//if(status != army_group_regiment_status::idle) {
-					//	continue;
-					//}
-
-					if(current_location == target_location) {
-						continue;
-					}
-					if(current_path.size() > 0) {
-						continue;
-					}
-
-					auto path = command::can_move_army(*this, local_player_nation, army, target_location);
-					if(path.empty()) {
-						// handle the case when there is no land path later
-					} else {
-						float weight = 0.f;
-						for(auto rg : world.army_get_army_membership(army)) {
-							weight += 3.0f;
-						}
-						if(weight < potential_size) {
-							command::move_army(*this, local_player_nation, army, path[0], false);
-							break;
-						} else {
-							std::array<dcon::regiment_id, command::num_packed_units> data;
-							int32_t i = 0;
-							data.fill(dcon::regiment_id{});
-							data[0] = current_regiment;
-							command::mark_regiments_to_split(*this, local_player_nation, data);
-							command::split_army(*this, local_player_nation, army);
-							break;
-						}
-					}
-				}
-			}
-		}
+		world.for_each_automated_army_group([&](dcon::automated_army_group_id item) {
```

Comment:
do we want this to be done every *frame*? Is there a reason that it couldn't be put in the on_update function, which is called every time the game state is updated (so at least once per in-game day)? That should be enough to let it respond to changes